### PR TITLE
Add tags to producer-only order cycle interface

### LINKED ISSUE #9248: Add Tag field to simple, producer-only order cycle interface
## Description
**- As a:** producer selling only their products (not hub)
**- On page:** when creating or editing an order cycle
**- I want to be able to do:** use tags, like hubs

## More context
We have the ability to setup tag rules showing/hiding OCs. However this can't be done for a producer only OC, because there is no field to type the tag. In a hub OC you see the tag field in the outgoing section, but this isn't there for simple OCs.

## Design proposal
screenshot from 2018-06-12 17-05-35

## How it's shown currently for hubs
image

## Acceptance Criteria & Tests

1. Login in as simple producer
2. I can add tags on the order cycle
3.  I can set up tags on enterprise settings, customers and shipping/payment methods
4. The rule defined works as expected.

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

Closes #9248

##### What has changed?
Add tag list editor to an order cycle for producer-only (not a hub) user:
Capture d’écran 2022-06-27 à 15 20 18

##### What has not changed?
Use this interface to specify tag rule:
Capture d’écran 2022-06-27 à 15 20 39

##### ... and, as a result:
Capture d’écran 2022-06-27 à 15 21 49

#### What should we test?
 - Login in as simple producer
 - I can add tags on the order cycle
 - I can set up tags on enterprise settings, customers and shipping/payment methods
 - The rule defined works as expected.

Also test the old system, ie. when it's a hub, and have the outgoing products interface with tags:
Capture d’écran 2022-06-28 à 15 06 44

#### Release notes

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by jibees** [comment]
Maybe needs some specs. @filipefurtad0 do have any idea on where I can put it?

**Comment by filipefurtad0** [comment]
This could be a good candidate:
`/spec/system/admin/order_cycles/simple_spec.rb`

On line 202 we're creating an order cycle as a hub, maybe a good place to check for tag-field introduced in this PR.

As on the functionality itself... Hum, I'm not sure. Tag functionality/filtering is covered here `spec/models/tag_rule/filter_order_cycles_spec.rb`, but in case we additionally need a system spec (for example, to check that we can hide an OC) then this one could be the place `spec/system/consumer/shopping/shopping_spec.rb`.

What do you think?

File: app/models/order_cycle.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@
 require 'open_food_network/scope_variant_to_hub'
 
 class OrderCycle < ApplicationRecord
+  acts_as_taggable
```
Comment (mkllnk) [inline app/models/order_cycle.rb:6]: Why don't we re-use the existing tagging functionality? Thoughts:

* Tagging order cycles is a simper data structure and it's easier to work with here.
* But we need to add more code to deal with order cycle tags.
* Simple order cycles still have exchanges we can tag.
* If we convert a simple order cycle to a complex one then the simple tags are suddenly hidden and can't be edited?

**Review by mkllnk — COMMENTED** [review]
Nice, this implementation is a lot simpler than I thought it would be. But I'm wondering if we can re-use the existing data structure. So a simple order cycle has only one outgoing exchange and we need to edit the tags of that one exchange. The rest of the code then stays the same and we don't need additional specs beyond checking that we can save tags via the simple page.

**Comment by jibees** [comment]
> Simple order cycles still have exchanges we can tag.
> a simple order cycle has only one outgoing exchange and we need to edit the tags of that one exchange.

nice catch, I didn't know. Will look at it! :pray:

**Review by mkllnk — APPROVED** [review]
Oh, wow, that's even less code than I thought. Great.

Shall we add a system spec though?

**Comment by jibees** [comment]
> Shall we add a system spec though?

I thought about it, but as it use the existing behavior, I was not sure. But maybe we could test the interface as an admin... (I was thinking adding it into `/system/consumer` but maybe its place is `/system/admin`)

**Comment by mkllnk** [comment]
We need to only test the new form. So when setting up a simple order cycle, type some tags in as well and check that they are saved.

**Review by mkllnk — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @jibees ,

Before the PR:

image

After:

image

Saving the tag on that the simple OC page works fine.

A quick sanity test:
- setting a hide OC tag
- setting an exception rule for customers
- tagging certain customers
- checking the OC is visible for the tagged customers, but not to others

All good here!
Merging :tada:

**Pull request merged**